lib/dash/content_protection.js

```diff
@@ -320,6 +320,50 @@ shaka.dash.ContentProtection = class {
     return ContentProtection.getLaurl_(rootElement);
   }
 
+
+  /**
+   * Gets a PlayReady initData from a content protection element
+   * containing a PlayReady Pro Object
+   *
+   * @param {shaka.dash.ContentProtection.Element} element
+   * @return {?Array.<shaka.extern.InitDataOverride>}
+   */
+  static getInitDataFromPro(element) {
+    const proNode = shaka.util.XmlUtils.findChildNS(
+        element.node, 'urn:microsoft:playready', 'pro');
+    if (!proNode) {
+      return null;
+    }
+    const fromBase64 = shaka.util.Uint8ArrayUtils.fromBase64;
+    const type = fromBase64(btoa('pssh'));
+    const systemId = fromBase64(btoa(element.schemeUri));
```

Comment:
This needs to be 16 bytes and should be binary.  For PlayReady it should be:

```js
const systemId = new Uint8Array([
    0x9a, 0x04, 0xf0, 0x79, 0x98, 0x40, 0x42, 0x86,
    0xab, 0x92, 0xe6, 0x5b, 0xe0, 0x88, 0x5f, 0x95,
]);
```

Since the `<mspr:pro>` element should only appear for PlayReady, you could just hard-code the system ID.